src/keycloak/chart/templates/uds-package.yaml

```
@@ -0,0 +1,84 @@
+apiVersion: uds.dev/v1alpha1
+kind: Package
+metadata:
+  name: keycloak
+  namespace: {{ .Release.Namespace }}
+spec:
+  network:
+    allow:
+      - description: "UDS Operator"
+        direction: Ingress
+        selector:
+          app.kubernetes.io/name: keycloak
+        remoteNamespace: pepr-system
+        remoteSelector:
+          app: pepr-uds-core-watcher
+        port: 8080
+        
+      # Temp workaround for any cluster pod
+      # @todo: remove this once cluster pods is a remote generated target
+      - description: "Keycloak backchannel access"
+        direction: Ingress
+        selector:
+          app.kubernetes.io/name: keycloak
+        remoteGenerate: Anywhere
+        port: 8080
+
+    expose:
+      - description: "remove private paths from public gateway"
+        host: sso
+        service: keycloak-http
+        selector:
+          app.kubernetes.io/name: keycloak
+        port: 8080
+        advancedHTTP:
+          match:
+            - name: redirect-welcome
+              uri:
+                exact: /
+            - name: redirect-admin
+              uri:
+                prefix: /admin
+            - name: redirect-master-realm
+              uri:
+                prefix: /realms/master
+            - name: redirect-metrics
+              uri:
+                prefix: /metrics
+          rewrite:
+            uri: "/realms/{{ .Values.realm }}/account"
+          headers:
+            request:
+              remove:
+                - istio-mtls-client-certificate
+              add:
+                istio-mtls-client-certificate: "%DOWNSTREAM_PEER_CERT%"
```

I think this can just be `set:` instead of `remove:` and `add:`

`set:` is documented as:
> Overwrite the headers specified by key with the given values

```suggestion
              set:
                istio-mtls-client-certificate: "%DOWNSTREAM_PEER_CERT%"
```

---

I ran into a problem with set and switched it--think when you didn't have a cert it was allowing any header you passed.